### PR TITLE
Extra environment variables of node

### PR DESCRIPTION
**Pull request by mhmxs**
Currently, there is no easy way to set CP/DP env vars that are not part of the StorageOSCluster CR.  Users can edit the ConfigMap directly but any changes will get overwritten during an upgrade.

 Instead, add a map[string]string to the StorageOSCluster CR that gets written to the ConfigMap.

### CODE REVIEW

**Comment by mhmxs** [comment]
I tested the following:
```
apiVersion: storageos.com/v1
kind: StorageOSCluster
metadata:
  name: example-storageoscluster
  namespace: "storageos"
spec:
  nodeExtraEnvironment:
    TEST_ENV: test-env
```

```
# kubectl exec -it -n storageos            storageos-node-84548 -- bash
Defaulted container "storageos" out of: storageos, csi-driver-registrar, csi-liveness-probe, init (init)
[root@kind-control-plane /]# echo $TEST_ENV
test-env
```

File: apis/v1/storageoscluster_types.go [hunk 1]
```diff
@@ -144,6 +144,10 @@ type StorageOSClusterSpec struct {
 
 	// Disable StorageOS scheduler extender.
 	DisableScheduler bool `json:"disableScheduler,omitempty"`
+
+	// NodeExtraEnvironment contains extra environment variables to pass
```
Comment (croomes) [inline apis/v1/storageoscluster_types.go:148]: nit: "NodeExtraEnvironment contains environment variables that are passed to StorageOS."

I'd be tempted to rename it to `ExtraEnvironment` or just `Environment`.  I can see us wanting to pass the same set to Node Manager or API Manager.

**Review by croomes — COMMENTED** [review]
Probably worth adding into the `04-update-config` kuttl test?

Otherwise LGTM - I'd expected this to be a lot more work :100:

File: apis/v1/storageoscluster_types.go (hunk 1 above)
Comment (mhmxs) [reply apis/v1/storageoscluster_types.go]: Makes sense, i renamed it to `Environment`.

**Review by croomes — APPROVED** [review]
Thanks @mhmxs , really glad to get this one in!

**Pull request merged**